src/backend/distributed/executor/citus_custom_scan.c

```diff
@@ -344,6 +344,10 @@ CitusBeginModifyScan(CustomScanState *node, EState *estate, int eflags)
 		 * Given that pruning is deferred always for INSERTs, we get here
 		 * !EnableFastPathRouterPlanner  as well.
 		 */
+		if (jobQuery->commandType == CMD_INSERT)
+		{
+			currentPlan->fastPathRouterPlan = true;
```

Comment:
here instead of setting `fastPathRouterPlan` to `true`, I think we could directly do below:

```c
Assert(jobQuery->commandType == CMD_INSERT || currentPlan->fastPathRouterPlan || !EnableFastPathRouterPlanner);
```

This is because, for example the insert with cte isn't actually fast path router, according to definition:

```c

/*
 * FastPathRouterQuery gets a query and returns true if the query is eligable for
 * being a fast path router query.
 * The requirements for the fast path query can be listed below:
 *
 *   - SELECT query without CTES, sublinks-subqueries, set operations
 *   - The query should touch only a single hash distributed or reference table
 *   - The distribution with equality operator should be in the WHERE clause
 *      and it should be ANDed with any other filters. Also, the distribution
 *      key should only exists once in the WHERE clause. So basically,
 *          SELECT ... FROM dist_table WHERE dist_key = X
 *      If the filter is a const, distributionKeyValue is set
 *   - All INSERT statements (including multi-row INSERTs) as long as the commands
 *     don't have any sublinks/CTEs etc
 */
```